main.go

```diff
@@ -7,29 +7,49 @@ import (
 )
 
 func main() {
-    boardSize := promptForBoardSize()
+    boardSize := 0
+    for boardSize == 0 {
+        boardSize = promptForBoardSize()
+    }
     game := NewGame(boardSize)
+    game.addBoardStars()
     game.printGame()
-    row := promptForRow()
-    col := promptForCol()
-    game.placePiece(row, col, game.currentMove)
-    game.changeTurn()
-    game.printGame()
+
+    gameOn := 1
+    for gameOn == 1 { //loop for multiple moves
+        piecePlaced := false
+        for piecePlaced == false { //loop for piece placement error
+            row := 0
+            for row == 0 {
+                row = promptForRow()
+            }
+            col := 0
+            for col == 0 {
+                col = promptForCol()
+            }
+            piecePlaced = game.placePiece(row, col, game.currentMove)
+        }
+        game.changeTurn()
+        game.printGame()
+    }
 }
 
 func promptForBoardSize() int {
-    fmt.Printf("Board size is nxn. Enter n: ")
+    fmt.Printf("Board size is nxn, and must be odd. Enter n: ")
     var inputtedBoardSize string
+    var findOdd int
     _, err := fmt.Scanf("%v", &inputtedBoardSize)
     if err != nil {
         log.Fatal(err)
     }
     boardSizeAsInt, err := strconv.ParseInt(inputtedBoardSize, 10, 8)
     if err != nil {
         fmt.Println("Error parsing input. Ensure board size is a valid integer.")
-        log.Fatal(err)
+        return int(0)
     }
-    return int(boardSizeAsInt)
+    findOdd = oddTester(int(boardSizeAsInt))
```

Comment:
here can we do
`findOdd := oddTester(int(boardSizeAsInt))`
and then remove the variable instantiation above on line 40?